Posts list - post upload trash/delete confirm dialog

mzorz opened the pull request:
Fixes #6306 

To test:
1. Start a new draft
2. include some media in it (preferrably some big images to make sure it takes some time to upload)
3. exit the editor
4. check the upload is progressing in the Posts list and the notifications
5. tap on "trash/delete" on the post being uploaded in the post list
6. the dialog is shown

![screenshot_20170725-103137](https://user-images.githubusercontent.com/6597771/28574615-dd33a9b2-7124-11e7-8510-4ffb89aad97c.png)


7. tap on OK
8. You'll see the post is hidden and the toast "Post deleted - UNDO" is shown.
9. wait a few seconds and verify the snackbar is gone and the notification is cleared as well when the snackbar is hidden.

cc @nbradbury @aforcier

aforcier commented:
I think the dialog should also have a 'cancel' option - but let's get some design input on this in general.

aforcier commented on WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java:
```diff
@@ -318,6 +318,20 @@ public static boolean isPostUploading(PostModel post) {
         return post != null && PostUploadHandler.isPostUploading(post);
     }
 
+    public static void abortPostAndRelatedMediaUpload(PostModel post) {
```
Could we refactor this to call `cancelQueuedPostUpload()` internally and drop some code duplication?

Also, what do you think of calling it `cancelQueuedPostUploadAndRelatedMedia()`, for a bit more consistency with that (similarly-behaving) method?

mzorz replied:
addressed in ee8e0b0

iamthomasbishop commented:
> I think the dialog should also have a 'cancel' option

Agreed. Perhaps the copy should probably read something like:

### Delete post?

Media in this post is currently uploading. Are you sure you'd like to delete this post?

[cancel] [delete]

Maybe the body copy could be simplified, but I'd defer to Editorial.

mzorz commented:
Added the `cancel` button and changed `OK` for `DELETE` as per @iamthomasbishop 's suggestion in a9d4256

![screenshot_20170727-145222](https://user-images.githubusercontent.com/6597771/28684674-af6e8c14-72db-11e7-8f25-9d78d1473142.png)


hi @iamthomasbishop! re: this:
> Media in this post is currently uploading. Are you sure you want to delete this post?

I'd prefer something more clear as this assertion doesn't imply the media will _also_ be cancelled (along with the post being deleted), as it's currently suggested in this PR. In this sense I'd stick with the current message, but waiting for Editorial seems a good choice 👍

michelleweber commented:
I think we can nix the bulky word "related," as it's pretty clear without it (and the user can see the media upload bar). I suggest:

Are you sure? Deleting this post will also cancel the media upload.

mzorz commented:
Awesome! Thank you @iamthomasbishop and @michelleweber 🙇  addressed this change in a7943b0

Ready to go @aforcier

aforcier commented:
Looks good! :shipit:
